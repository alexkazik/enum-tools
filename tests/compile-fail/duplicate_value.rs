#![no_implicit_prelude]

use ::enum_tools::EnumTools;

#[derive(EnumTools, Debug, Clone, Copy, Eq, PartialEq)]
#[repr(i8)]
pub(crate) enum EG {
    #[enum_tools(rename = "A*")]
    A = 0,
    B = 9,
    C = 2,
    D = 9,
}

fn main() {}
