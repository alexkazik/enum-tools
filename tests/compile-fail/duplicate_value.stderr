error: Duplicate value
  --> tests/compile-fail/duplicate_value.rs:12:5
   |
12 |     D = 9,
   |     ^

error[E0081]: discriminant value `9` assigned more than once
  --> tests/compile-fail/duplicate_value.rs:7:1
   |
7  | pub(crate) enum EG {
   | ^^^^^^^^^^^^^^^^^^
...
10 |     B = 9,
   |         - `9` assigned here
11 |     C = 2,
12 |     D = 9,
   |         - `9` assigned here
